File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target)
    {
        map<int, int> comp;

        for (int i = 0; i < nums.size(); i += 1)
        {
            auto pos = comp.find(target - nums[i]);
            if (pos != comp.end())
                return {pos->second, i};
            comp[nums[i]] = i;
        }
        return {-1, -1};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        HashMap<Integer, Integer> hash = new HashMap<>();
        int flag = 0;
        for (int i=0; i<nums.length; i++)
        {
            int b = target - nums[i]; //a + b = target, target - a = b
            if (hash.containsKey(b))
            {
                //if the map contains it 
                flag = 1;
                return new int[] { i, hash.get(b) };
            }
            hash.put(nums[i], i);
        }

        return new int[] {};
    }
}